packages/flutter/lib/src/painting/text_style.dart

```diff
@@ -230,10 +232,13 @@ class TextStyle extends Diagnosticable {
     this.decoration,
     this.decorationColor,
     this.decorationStyle,
+    String debugLabel,
     String fontFamily,
     String package,
   }) : fontFamily = package == null ? fontFamily : 'packages/$package/$fontFamily',
-       assert(inherit != null);
+       debugLabel = debugLabel == null ? _kDefaultDebugLabel : debugLabel,
+       assert(inherit != null),
+       assert(debugLabel == null || debugLabel.length <= 100);
```

Comment:
I'd avoid the maximum length. It's likely that we'll accidentally exceed it at some point when creating a lerp of a lerp of a lerp or some crazy thing.

Reply:
(...but need to see the whole length because it's meaningful in some way to what we're debugging, or something)

Reply:
I guess maybe you added this because in regular operation we generate 1000-long messages? I defer to your judgement on this. If we do generate ridiculous chains in normal operation, let me know, though. That seems surprising.